pallets/author-mapping/src/tests.rs

```diff
@@ -164,7 +164,20 @@ fn registered_author_cannot_be_cleared_by_non_owner() {
 		})
 }
 
-// Registered author cannot be stolen by someone else registering over top
+#[test]
+fn registered_author_cannot_be_overwritten() {
```

Comment:
SO the current behavior is that an author id can only be registered once?
Two problems with that approach imho:
- what if the malicious user registers the author id forst?
- what if the author id wants to change their associated accountid ?

Reply:
> what if the malicious user registers the author id forst?

Yes this is a valid complaint as we both agreed over in https://github.com/PureStake/moonbeam/pull/413#issuecomment-847984145. We need to address it in a follow up. I agree 100%

> what if the author id wants to change their associated accountid ?

This is not supported. `AccountIds` are the things that participate in staking, hold funds, and stick around for a while. `AuthorIds` are short-lived should-be-rotated-regularly session keys that sign blocks. A staked collator `AccountId` will typically go through many `AuthorIds` if it is staked for a long time. An `AuthorId` should never be reused _especially_ for a different account Id.